### PR TITLE
Move dmd.aliasthis.resolveAliasThis to dsymbolsem

### PR DESCRIPTION
**Pull request by RazvanN7**

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]
otherwise looks good

File: compiler/src/dmd/aliasthis.d [hunk 1]
```diff
@@ -13,17 +13,12 @@
 
 module dmd.aliasthis;
 
-import core.stdc.stdio;
```
[the remark this answers is not in the document]
Comment (dkorpel) [reply compiler/src/dmd/aliasthis.d:16]: There's a commented printf in this file:

```D
//printf("+isRecursiveAliasThis(att = %s, t = %s)\n", att ? att.toChars() : "null", t.toChars());
```

[the remark this answers is not in the document]
Comment (RazvanN7) [reply compiler/src/dmd/aliasthis.d:16]: Yeah, but that function will be moved to mtype.d

[the remark this answers is not in the document]
Comment (RazvanN7) [reply compiler/src/dmd/aliasthis.d:16]: Anyway, I concede.

[the remark this answers is not in the document]
Comment (RazvanN7) [reply compiler/src/dmd/aliasthis.d:16]: Re-added.

**Pull request merged**